### DISCUSSION TITLE
Any ways to make the simulation faster?

### DISCUSSION THREAD

**Discussion by khushit-shah** [ChampSim/ChampSim#394] [Q&A] [answered]
Hii, I'm am testing a new Branch Predictor and I'm only interested in MPKI, IPC and Prediction Accuracy metrics, the current configuration file is as follows:
```
{
    "branch_predictor": "my-predictor"
}
```
Is there any way to make the simulations faster? I was thinking about immediate memory access, or making all modules other than branch prediction behave same as they do in warmup stage...

**Comment by ngober** [comment, marked as answer]
What exactly would you hope that your faster simulations would model? For example, you suggest immediate memory accesses. In ChampSim's warmup phase, most latencies are reduced to zero. This has tremendous implications for IPC, to the point where we entirely discard the stats for the warmup phase. If you want to do that, then your configuration file could look like

    {
        "L1I": {
            "hit_latency": 0,
            "fill_latency": 0
        }
        // etc for the rest of the caches

        "decode_latency": 0,
        // etc for the rest of the core
    }
